File: frontend/src/interfaces/Transaction.ts
import { IResponse } from './APIResponse';

export interface ITransaction {
  id: string;
  type: 'income' | 'expense';
  from: string;
  sender_name: string;
  to: string;
  recipient_name: string;
  amount: string;
  source_of_funds: string;
  description: string;
  date: string;
}

export interface ITransactionList {
  total_item: number;
  transactions: ITransaction[];
}

export interface ITransactionResponse extends IResponse {
  data: ITransactionList;
}

File: backend/entity/transaction.go
package entity

import (
	"time"

	"github.com/shopspring/decimal"
)

type Transaction struct {
	Id                 int
	Type               string
	SenderWalletNum    string
	SenderName         string
	RecepientWalletNum string
	RecipientName      string
	Amount             decimal.Decimal
	SourceOfFunds      string
	Description        string
	Date               time.Time
}

type TransactionList struct {
	TotalItem    int
	Page         string
	Transactions []Transaction
}
